multicluster/controllers/multicluster/memberclusterannounce_controller.go

```diff
@@ -114,38 +108,34 @@ func (r *MemberClusterAnnounceReconciler) Reconcile(ctx context.Context, req ctr
 		klog.V(2).InfoS("Reset lastUpdateTime", "cluster", memberAnnounce.ClusterID)
```

Comment:
After reading the code more, I wonder should we do the following two:

1. Merge timerData and memberStatus to one map (and call it memberStatusMap)
2. Move this block to addMember() (and rename the func to sth like addUpdateMemberStatus())

What you think?

Reply:
Yes, I am thinking it should work, but If we'd like to include this PR to 1.8, I feel it's better to has a follow up PR to rethink about the whole status part, what's your suggestion?

Reply:
I am fine to do another PR, but I do feel we should clean up the code. The current implementation is unnecessarily complex, hard to follow and may include bugs due to that.

Reply:
Sure, I feel we need rethink the design including ClusterClaim/ClusterSet and status.